src/lucky_migrator/create_table_statement.cr

```diff
@@ -41,6 +57,21 @@ class LuckyMigrator::CreateTableStatement
     end
   end
 
+  # Generates raw sql for adding an index to a table column. Accepts 'unique' and 'using' options.
+  def add_index(column : String | Symbol, unique = false, using : String | Symbol = "btree")
+    raise "index type '#{using}' not supported" unless ALLOWED_INDEX_TYPES.includes?(using.to_s)
```

Comment:
Maybe this should be moved into `CreateIndexStatement`? That way anything that uses it gets the error message for free without needing to duplicate it

Reply:
Do you mean the Allowed_index types, or the add_index method as a whole?

Reply:
Oh yeah I wasn't clear there. I mean just the `ALLOWED_INDEX_TYPES` and the raise statement. That could be done in `CreateIndexStatement` I think

Reply:
K thanks for the clarification, you're right that works much better there